src/buffer.c

```diff
@@ -91,6 +91,36 @@ BUFFER_HANDLE BUFFER_create(const unsigned char* source, size_t size)
     return (BUFFER_HANDLE)result;
 }
 
+BUFFER_HANDLE BUFFER_create_with_size(size_t buff_size)
+{
+    BUFFER* result;
+    if (buff_size == 0)
```

Comment:
>  if (buff_size == 0) [](start = 2, length = 21)

malloc allows zero sizes, BUFFER_create also does, so this should also allow it.